Scripts/script-DBotPredictPhishingEvaluation.yml

```diff
@@ -0,0 +1,169 @@
+commonfields:
+  id: DBotPredictPhishingEvaluation
+  version: -1
+name: DBotPredictPhishingEvaluation
+script: |+
+  import pandas as pd
+  from sklearn import metrics
+  import sys
+
+  # disable warnings
+  import warnings
+  warnings.filterwarnings('always')
+
+  def get_threshold_precision(df, threshold, predicted_col, original_col, threshold_col):
+      skipped_count = len(df[df[threshold_col] < threshold])
+      if skipped_count == len(df):
+          precantage_skipped = 100
+          precision_score = 0
+          recall_score = 0
+          f1_score = 0
+      else:
+          precision_score = metrics.precision_score(df[df[threshold_col] >= threshold][original_col], df[df[threshold_col] >= threshold][predicted_col], average='micro') * 100
+          recall_score = metrics.recall_score(df[df[threshold_col] >= threshold][original_col], df[df[threshold_col] >= threshold][predicted_col], average='micro') * 100
+          f1_score = metrics.f1_score(df[df[threshold_col] >= threshold][original_col], df[df[threshold_col] >= threshold][predicted_col], average='micro') * 100
+          precantage_skipped = (float(skipped_count) / len(df)) * 100
+
+      return {
+          'Threshold': threshold,
+          'Precision': '%.2f%%' % precision_score,
+          'Recall': '%.2f%%' % recall_score,
+          'Below Threshold': '%.2f%%' % precantage_skipped,
+          'F1 Score': '%.2f%%' % f1_score,
+      }
+
+  entries = demisto.executeCommand('DBotPreparePhishingData', demisto.args())
+  if isError(entries[0]):
+      demisto.results(entries[0]['Contents'])
+      sys.exit(0)
+  else:
+      entries = [e for e in entries if e['ContentsFormat'] == formats['json']]
+      labeled_data = entries[0]['Contents']
+  if len(labeled_data) == 0:
+      return_error("There is not data to evaluate")
+  max_number_of_incidents = int(demisto.args()['maxNumberOfIncidents'])
+  if len(labeled_data) == max_number_of_incidents:
+      demisto.log('Results from incidentsQuery has reached to the maximum (maxNumberOfIncidents=%d). Consider increase maxNumberOfIncidents to get better results' % max_number_of_incidents)
+  demisto.log("Evaluate %d samples" % len(labeled_data))
+  res = demisto.executeCommand('DBotPredictTextLabel', {
+      'inputText': map(lambda x: x['txt'], labeled_data),
+      'modelListName': demisto.args()['modelListName']
+  })
+  if isError(res[0]):
+      return_error(res[0]['Contents'])
+
+
+  predictions = res[0]['Contents']
+
+  df = pd.DataFrame.from_dict(predictions)
+  df.loc[:, 'Original'] = pd.Series(map(lambda x: x['label'], labeled_data))
+
+  result = []
+  for i in xrange(60, 100, 5):
+      if i < 80:
+          if i % 10 == 0:
```

Comment:
Why do we skip the 5?

Reply:
it's too much. I will split it to 2 ranges... will be simple to read